src/coreclr/jit/targetamd64.h

```diff
@@ -563,4 +563,9 @@
   #define RBM_STACK_PROBE_HELPER_TRASH RBM_RAX
 #endif // !UNIX_AMD64_ABI
 
+  // #define SWIFT_SUPPORT
```

Comment:
Does this mean that the code from this PR won't be active now? If so, what is the plan for making it available for macOS?

Reply:
Yes, it's inactive for now. I don't mind enabling it now -- it would be nice to ensure these changes don't affect existing behavior. @kotlarmilos @jakobbotsch any preference for turning this on now?

> If so, what is the plan for making it available for macOS?

I can update `SWIFT_SUPPORT` to be enabled only when targeting macOS on arm64/x64.

Reply:
I think we should enable it here (although I assume you'll need to mark some of the `SwiftSelf` tests as disabled on coreclr for now).

Note that we don't build any JIT specifically for macOS, so enabling `SWIFT_SUPPORT` only on macOS is not that simple (requires runtime checks). I think you should just enable `SWIFT_SUPPORT` unconditionally on arm64/x64, and we can look at what the impact of that is.

Reply:
Sounds good, I've enabled `SWIFT_SUPPORT` in the JIT.

> I think we should enable it here (although I assume you'll need to mark some of the SwiftSelf tests as disabled on coreclr for now).

I think all of the Swift interop tests are disabled for now. Should I enable the SwiftErrorHandling tests with this PR too? I think we'd also have to allow the Swift calling convention in the VM (`dllimport.cpp`, etc.) for those to run correctly.

Reply:
That does look suspicious. It sounds fine to me to comment the free call for now. @kotlarmilos ?

Reply:
The `NativeMemory.Free` is a thin wrapper over the C free API. We experienced a similar issue in Mono, where context registers were not correctly restored after the P/Invoke. If this is a blocker, you can temporarily disable it but please create a tracking issue, so we can address it at some point.

Reply:
Is there any guarantee that the same allocator is used on the Swift side? I don't think so. It seems like the right fix would be to pinvoke back to Swift to free up the memory.